### ISSUE TITLE
Strange behavior of the Finding series by label matchers API

### ISSUE THREAD

**Issue by DMANSUTTI** [VictoriaMetrics/VictoriaMetrics#250]
**Describe the bug**
While using the finding series by label matchers API, if the first request on a time series is within a time range that does not contain any data, all following requests fail even for time range with actual data.

**To Reproduce**

Create a metric error_code through the influxDB protocol line API with some data beyond to the timestamp value 1574758291 but not up to value 1575263091.

The sequence that works is the following (after a restart of VictoriaMetrics):

1. `curl -d "match[]=error_code&start=1574758291&end=1575363091" -X POST http://localhost:8428/api/v1/series`
`{"status":"success","data": [{"__name__":"error_code","device":"35881608","code":"8.2.1-3.10","operator":"901","profile":"profile1"}...`
2. `curl -d "match[]=error_code&start=1575263091&end=1575363091" -X POST http://localhost:8428/api/v1/series`
`{"status":"success","data":[]}`
3. `curl -d "match[]=error_code&start=1574758291&end=1575363091" -X POST http://localhost:8428/api/v1/series`
`{"status":"success","data":[{"__name__":"error_code","device":"35881608","code":"8.2.1-3.10","operator":"901","profile":"profile1"}`

But if we start with the one with no data, then the one with data also fails. After restarting VictoriaMetrics, try the following sequence to reproduce the issue.

1. `curl -d "match[]=error_code&start=1575263091&end=1575363091" -X POST http://localhost:8428/api/v1/series`
`{"status":"success","data":[]}`
2. `curl -d "match[]=error_code&start=1574758291&end=1575363091" -X POST http://localhost:8428/api/v1/series`
`{"status":"success","data":[]}`

**Expected behavior**

The response should return the corresponding data whatever the response to the initial request was.

** Version **
The current master branch.

**Comment by valyala** [comment]
Thanks for the bug report. Unfortunately the issue couldn't be reproduced locally.

Could you provide the output for the following query for `/api/v1/export` after the corresponding query to `/api/v1/series` returns empty result:
```
curl -d "match[]=error_code&start=1574758291&end=1575363091" -X POST http://localhost:8428/api/v1/export
```

This could be related to the cache. Could you also stop VcitoriaMetrics, remove `<-storageDataPath>/cache`, then start it again and verify whether the query to `/api/v1/series` returns empty result?

Could you also check VictoriaMetrics logs? Probably something goes wrong during inserting the data or during the queries.

**Comment by DMANSUTTI** [comment]
Hello,

Please find hereafter the additional requested information.

- Export API: it does not display any data (while it does with the working sequence)
- Remove the cache: it does not change anything to the non-working sequence
- Logs: I do not have anymore the log of the data insertion at it was done a couple of weeks ago

I will relaunch the data insertion and check for log errors and redo the test tomorrow.

For your information I insert 1 year of data per 1h steps encompassing 8 metrics, each one having a set of labels which approximately leads to 60000 time series.

**Comment by valyala** [comment]
Thanks for the additional information!

Make sure you use the correct `-retentionPeriod` command-line flag value. It should be 12 or higher for 1 year of data.

**Comment by DMANSUTTI** [comment]
I installed the latest official version (v1.30.4), cleaned the time series and did the data insertion. There was no error during the data insertion and then I waited a couple of hours to be sure to have time range with no data and re-did the previously non working sequence.

The sequence is now working and I do not see the strange behavior anymore. As I am not sure how and why I ended up with this strange behavior and as I am not able to investigate further I think we can close the issue. I will anyway monitor the evolution in the coming days to see if something triggers the behavior.

Tell me if you want me to do other actions for this issue.

Thanks for your help.
